shared/src/sources/balancer/pool_fetching.rs

```diff
@@ -105,19 +106,42 @@ impl BalancerPoolEvaluating for WeightedPool {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct AmplificationParameter {
+    factor: U256,
+    precision: U256,
+}
+
+impl AmplificationParameter {
+    pub fn new(factor: U256, precision: U256) -> Self {
+        Self { factor, precision }
+    }
+
+    /// This is the format used to pass into smart contracts.
+    pub fn as_u256(&self) -> Option<U256> {
+        self.factor.checked_mul(self.precision)
```

Comment:
I think the format is `self.factor` not `self.factor * self.precision`.

Reply:
That could very well be! Thanks.